tests/network/vmi_passt.go

```diff
@@ -0,0 +1,220 @@
+					serverIP := libnet.GetVmiPrimaryIPByFamily(serverVMI, ipFamily)
+					err := libnet.PingFromVMConsole(clientVMI, serverIP)
+					if err != nil {
+						return err
+					}
+
+					By("Connecting from the client VM")
+					err = console.SafeExpectBatch(clientVMI, checkConnectionToServer(serverIP, tcpPort, true), 30)
+					if err != nil {
+						return err
+					}
+
+					return nil
+				}
+
+				startServerVMI := func(ports []v1.Port) {
+					serverVMI = libvmi.NewCirros(
+						libvmi.WithInterface(libvmi.InterfaceDeviceWithPasstBinding(ports...)),
+						libvmi.WithNetwork(v1.DefaultPodNetwork()),
+						withPasstExtendedResourceMemory(ports...),
+					)
+
+					serverVMI, err = virtClient.VirtualMachineInstance(util.NamespaceTestDefault).Create(serverVMI)
+					Expect(err).ToNot(HaveOccurred())
+					serverVMI = tests.WaitForSuccessfulVMIStartIgnoreWarnings(serverVMI)
+					Expect(console.LoginToCirros(serverVMI)).To(Succeed())
+				}
+
+				BeforeEach(func() {
+					clientVMI = libvmi.NewCirros(
+						withPasstDefaultPort(),
+						libvmi.WithNetwork(v1.DefaultPodNetwork()),
+					)
+
+					clientVMI, err = virtClient.VirtualMachineInstance(util.NamespaceTestDefault).Create(clientVMI)
+					Expect(err).ToNot(HaveOccurred())
+					clientVMI = tests.WaitForSuccessfulVMIStartIgnoreWarnings(clientVMI)
+					Expect(console.LoginToCirros(clientVMI)).To(Succeed())
+				})
+
+				DescribeTable("TCP", func(ports []v1.Port, tcpPort int, ipFamily k8sv1.IPFamily) {
+					libnet.SkipWhenClusterNotSupportIPFamily(virtClient, ipFamily)
+
+					startServerVMI(ports)
+
+					By("starting a TCP server")
+					tests.StartTCPServer(serverVMI, tcpPort, console.LoginToCirros)
+
+					Expect(verifyClientServerConnectivity(clientVMI, serverVMI, tcpPort, k8sv1.IPv4Protocol)).To(Succeed())
+
+					if len(ports) != 0 {
+						By("starting a TCP server on a port not specified on the VM spec")
+						vmPort := int(ports[0].Port)
+						serverIP := libnet.GetVmiPrimaryIPByFamily(serverVMI, ipFamily)
+
+						tests.StartTCPServer(serverVMI, vmPort+1, console.LoginToCirros)
+
+						By("Connecting from the client VM to a port not specified on the VM spec")
+						Expect(console.SafeExpectBatch(clientVMI, checkConnectionToServer(serverIP, tcpPort+1, true), 30)).To(Not(Succeed()))
+					}
+				},
+					Entry("with a specific port number [IPv4]", []v1.Port{{Name: "http", Port: 8080, Protocol: "TCP"}}, 8080, k8sv1.IPv4Protocol),
+					Entry("without a specific port number [IPv4]", []v1.Port{}, 8080, k8sv1.IPv4Protocol),
+					Entry("with a specific port number [IPv6]", []v1.Port{{Name: "http", Port: 8080, Protocol: "TCP"}}, 8080, k8sv1.IPv6Protocol),
+					Entry("without a specific port number [IPv6]", []v1.Port{}, 8080, k8sv1.IPv6Protocol),
+				)
+			})
+
+			It("[outside_connectivity]should be able to reach the outside world [IPv4]", func() {
+				libnet.SkipWhenClusterNotSupportIpv4(virtClient)
+				ipv4Address := "8.8.8.8"
```

Comment:
Wouldn't it be enough to run a http servier in a container within the cluster, so that we don't need to reach external IP like 8.8.8.8?